rcl/src/rcl/logging.c

```diff
@@ -76,6 +77,14 @@ rcl_logging_configure_with_output_handler(
   if (log_levels) {
     default_level = (int)log_levels->default_logger_level;
     rcutils_logging_set_default_logger_level(default_level);
+    if (RCUTILS_LOG_SEVERITY_UNSET != default_level) {
+      rmw_ret_t rmw_status = rmw_set_log_severity((rmw_log_severity_t)default_level);
```

Comment:
What you are doing here makes a certain sense; if we are changing the default logging level, we should change it everywhere.

On the other hand, I'm not sure what semantic we really want `--log-level` to have.  As it stands right now, the semantic is more-or-less "core ROS 2 code".  If we add this code in, it becomes "core ROS 2 code + RMW backend code".  I'm not entirely sure that's what we want to do.

I think this would be helped by spelling out what the log-levels mean in a more rigorous way, probably as part of https://github.com/ros2/design/issues/314 .

Reply:
I understand. The fact that rmw implementations don't support the "unset" logging level/state definitely hightlights this.

Are there any plans to tackle that soon-ish after Galactic? I can't commit to doing ros2/design#314, but I can probably help, and I can certainly implement whatever is in the design document for this.

Reply:
> Are there any plans to tackle that soon-ish after Galactic?

The honest answer is that I don't know right now.  We've been concentrating on Galactic, so we don't have a roadmap for H-Turtle yet.  I do hope it is something that the ROS 2 community gets to for H-Turtle, as I think it is important.

Reply:
Got it. I'll see what I can do (in a few weeks) then.

I'll leave this draft here for now.